### ISSUE TITLE
TZDate class toISOString method returns incorrect date string after adding minutes

### ISSUE THREAD

**Issue by ShinyaKato** [date-fns/tz#16]
### Environment

* `date-fns` 4.1.0
* `@date-fns/tz` 1.1.2

### Summary

Thanks for supporting time zones!
I tried using it right away, but I found a strange behavior with the `toISOString` method, so I'm reporting it.

`toISOString` method in the `TZDate` class returns an incorrect date string after adding minutes.

### Steps to Reproduce

1. Create `TZDate` instance.
2. Use the `addMinutes` function to add 30 minutes.
3. Call `toISOString` method.

```js
import { TZDate } from "@date-fns/tz";
import { addMinutes } from "date-fns";

const original = new TZDate("2024-09-26T10:00:00.000+09:00", "Asia/Tokyo");
const added = addMinutes(original, 30);
console.log(added.toISOString());
```

### Expected Result

`added.toISOString()` should returns the following string:

```
2024-09-26T10:30:00.000+09:00
```

### Actual Result

`added.toISOString()` method returns an incorrect date string.

```
2024-09-26T10:00:00.000+09:00
```

The result of `addMinutes` does not seem to be reflected.

### Additional Information

When using the `format` function, the correct result is returned.

```js
import { TZDate } from "@date-fns/tz";
import { addMinutes, format } from "date-fns";

const original = new TZDate("2024-09-26T10:00:00.000+09:00", "Asia/Tokyo");
const added = addMinutes(original, 30);
console.log(format(added, "yyyy-MM-dd'T'HH:mm:ss.SSSXXX"));
// "2024-09-26T10:30:00.000+09:00"
```

**Comment by johnny-T** [comment]
The same happens for .toString() method too...

**Comment by codeflows** [comment]
[Noticed the same](https://github.com/date-fns/tz/issues/24). It seems like the `TZDate.setTime` implementation used by `addMinutes` is misbehaving, while e.g. `startOfDay` works (it uses `TZDate.setHours`).

**Comment by kossnocorp** [comment]
Thank you for letting me know, I'm working on it.

**Comment by kossnocorp** [comment]
Thank you for the patience, the fix is out with `@date-fns/tz@1.2.0`
